includes/documents/abstract-wcpdf-order-document.php

```diff
@@ -535,16 +535,22 @@ public function is_allowed(): bool {
 			}
 		}
 
-		// Check document prerequisites
+		return apply_filters( 'wpo_wcpdf_document_is_allowed', $allowed, $this );
+	}
+
+	public function do_prerequisites_exist(): bool {
```

Comment:
Do we need this function? Can't we use only the `get_prerequisite_documents()`?

Reply:
I thought the suggestion [here](https://github.com/wpovernight/woocommerce-pdf-invoices-packing-slips/pull/757#discussion_r1568778838) was to create a new method and move checking prerequisites condition to it from the `is_allowed()` method.

Maybe I didn't understand your suggestion correctly.

Reply:
I'm not sure if the `get_prerequisite_documents()` existed at the time of my comment, but what I wanted to say is having one function to handle all the `prerequisite`, not having multiple functions. Sorry if it was not clear.

But let me know your opinion.

Reply:
I think it would be good to keep the `get_prerequisite_documents()` method as it is, since it acts as a getter for prerequisite documents.
I agree that the `do_prerequisites_exist()` method didn't make improvements and we just ended up with another method.
Maybe it's better to revert the recent change and go back to the previous implementation?

Reply:
Ok, let's do that.